lib/puppet/type/file.rb

```diff
@@ -379,6 +379,11 @@ def self.title_patterns
 
     self.warning "Possible error: recurselimit is set but not recurse, no recursion will happen" if !self[:recurse] and self[:recurselimit]
 
+    if @parameters[:content] and @parameters[:content].actual_content
+      # Now that we know the checksum, update content (in case it was created before checksum was known).
+      @parameters[:content].value = @parameters[:checksum].sum(@parameters[:content].actual_content)
```

Comment:
On more consideration, I'll explore moving re-initialization to the Content class.